istio-2-mixin/panels.libsonnet

```
@@ -0,0 +1,1061 @@
+        + stat.options.withGraphMode('none')
+        + stat.standardOptions.color.withMode('thresholds')
+        + stat.standardOptions.thresholds.withSteps([
+          stat.thresholdStep.withColor('text')
+          + stat.thresholdStep.withValue(null),
+          stat.thresholdStep.withColor('super-light-green')
+          + stat.thresholdStep.withValue(1),
+        ]),
+      workloadEntries:
+        commonlib.panels.generic.stat.base.new(
+          'Workload entries',
+          targets=[t.workloadEntryCount],
+          description='Number of workload entries in the Istio system.',
+        )
+        + stat.options.withGraphMode('none')
+        + stat.standardOptions.color.withMode('thresholds')
+        + stat.standardOptions.thresholds.withSteps([
+          stat.thresholdStep.withColor('text')
+          + stat.thresholdStep.withValue(null),
+          stat.thresholdStep.withColor('super-light-green')
+          + stat.thresholdStep.withValue(1),
+        ]),
+
+      vCPUUsage:
+        commonlib.panels.generic.timeSeries.base.new(
+          'vCPU usage',
+          targets=[
+            t.istiodCPUUsage,
+            t.gatewayCPUUsage,
+            t.proxyCPUUsage,
+          ],
+          description='vCPU usage for various components of the Istio system.',
+        )
+        + timeSeries.options.legend.withPlacement('right')
+        + timeSeries.fieldConfig.defaults.custom.withStackingMixin({
+          group: 'A',
+          mode: 'normal',
+        })
+        + timeSeries.standardOptions.withUnit('percentunit'),
+      openFileDescriptors:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Open file descriptors',
+          targets=[
+            t.istiodOpenFileDescriptors,
+            t.gatewayOpenFileDescriptors,
+            t.proxyOpenFileDescriptors,
+          ],
+          description='Number of open file descriptors for various components of the Istio system.',
+        )
+        + timeSeries.options.legend.withPlacement('right')
+        + timeSeries.fieldConfig.defaults.custom.withStackingMixin({
+          group: 'A',
+          mode: 'normal',
+        }),
+      virtualAndResidentMemory:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Virtual & resident memory',
+          targets=[
+            t.istiodVirtualMemory,
+            t.istiodResidentMemory,
+            t.gatewayVirtualMemory,
+            t.gatewayResidentMemory,
+            t.proxyVirtualMemory,
+            t.proxyResidentMemory,
+          ],
+          description='Available virtual memory compared to the resident memory for the various components of the Istio system.',
+        )
+        + timeSeries.options.legend.withPlacement('right')
+        + timeSeries.fieldConfig.defaults.custom.withScaleDistributionMixin({
+          log: 2,
+          type: 'log',
+        })
+        + timeSeries.standardOptions.withUnit('bytes'),
+      heapMemory:
+        commonlib.panels.generic.timeSeries.base.new(
```

```suggestion
        commonlib.panels.memory.timeSeries.usageBytes.new(
```
You could try memory specific panel instead:

---

Changed